File: VB/TextEditPasswordStrengthBar/TextEditPwdStrength/RepositoryItemTextEditPwdStrength.vb
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Controls

Namespace TextEditPasswordStrengthBar

    'The attribute that points to the registration method 
    <UserRepositoryItem("RegisterTextEditPwdStrength")>
    Friend Class RepositoryItemTextEditPwdStrength
        Inherits RepositoryItemTextEdit

        Private Shared ReadOnly editValueChanging As Object = New Object()

        ' Static constructor should call registration method
        Shared Sub New()
            Call RegisterTextEditPwdStrength()
        End Sub

        Public Const TextEditPwdStrengthName As String = "TextEditPwdStrength"

        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return TextEditPwdStrengthName
            End Get
        End Property

        Public Shared Sub RegisterTextEditPwdStrength()
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(TextEditPwdStrengthName, GetType(TextEditPwdStrength), GetType(RepositoryItemTextEditPwdStrength), GetType(TextEditPwdStrengthViewInfo), New TextEditPwdStrengthPainter(), True))
        End Sub
    End Class
End Namespace

File: CS/TextEditPasswordStrengthBar/TextEditPwdStrength/RepositoryItemTextEditPwdStrength.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Controls;

namespace TextEditPasswordStrengthBar
{
    //The attribute that points to the registration method 
    [UserRepositoryItem("RegisterTextEditPwdStrength")]
    class RepositoryItemTextEditPwdStrength : RepositoryItemTextEdit
    {
        private static readonly object editValueChanging = new object();

        // Static constructor should call registration method
        static RepositoryItemTextEditPwdStrength() { RegisterTextEditPwdStrength(); }

        public const string TextEditPwdStrengthName = "TextEditPwdStrength";
        public override string EditorTypeName { get { return TextEditPwdStrengthName; } }

        public static void RegisterTextEditPwdStrength()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(
                TextEditPwdStrengthName, typeof(TextEditPwdStrength), typeof(RepositoryItemTextEditPwdStrength),
                typeof(TextEditPwdStrengthViewInfo), new TextEditPwdStrengthPainter(), true));
        }
    }
}
